common/types.go

```diff
@@ -57,6 +59,38 @@ type EncryptedPayloadHash [EncryptedPayloadHashLength]byte
 // Using map to enable fast lookup
 type EncryptedPayloadHashes map[EncryptedPayloadHash]struct{}
 
+func (h *EncryptedPayloadHash) MarshalJSON() (j []byte, err error) {
+	return json.Marshal(h.ToBase64())
+}
+
+func (h *EncryptedPayloadHash) UnmarshalJSON(j []byte) (err error) {
+	var ephStr string
+	json.Unmarshal(j, &ephStr)
```

Comment:
catch returned error
```suggestion
	err = json.Unmarshal(j, &ephStr)
	if err != nil {
		return
	}
```